docs/references/auth_code_samples.rst

```diff
@@ -0,0 +1,171 @@
+    'expires_in': 3600,
+    'refresh_token': 'm8iXhVlGABu52xFxVFj5rAz8xSjsRq',
+    'scope': 'read write email profile',
+    'token_type': 'JWT'}
+
+.. note:: The client type must be ``public`` for this to work.
+
+.. _JWT from application:
+
+Get a JWT with a client_id and client_secret
+********************************************
+
+.. code-block::
+
+   import base64
+   import requests
+
+   from pprint import pprint
+
+   client_id = "ukbclQB8aPh7hgsy8ifPXkPf7fRqgUq1w21f2YZa"
+   # Note this should actually be secert and probably not in your code but
+   # provided here in the example
+   client_secret = "xkN0BJ19q9Jk8UPUppEtC1xe4764c81ioFtlegvokbmnAC7CFCT5gG1Og5nnFmCNc3NHNhUwWWDRVcBfnLSZ4xAlEmSePzfkFtLE06cwR1MuSc0gx9LUEjRrTs3j2vgK"
+
+   credential = f"{client_id}:{client_secret}"
+   encoded_credential = base64.b64encode(credential.encode("utf-8")).decode("utf-8")
+
+
+   headers = {"Authorization": f"Basic {encoded_credential}", "Cache-Control": "no-cache"}
+   data = {"grant_type": "client_credentials", "token_type": "jwt"}
+
+   token_request = requests.post(
+       "http://lms.example.com/oauth2/access_token", headers=headers, data=data
+   )
+
+   pprint(token_request.json())
+
+.. code-block::
+   :caption: Output
+
+   {'access_token': 'eyJhbGciOiJIUzI1NiJ9.eyJhdWQiOiAibG1zLWtleSIsICJleHAiOiAxNjkyMjExNjM4LCAiZ3JhbnRfdHlwZSI6ICJjbGllbnQtY3JlZGVudGlhbHMiLCAiaWF0IjogMTY5MjIwODAzOCwgImlzcyI6ICJodHRwOi8vbG9jYWxob3N0OjE4MDAwL29hdXRoMiIsICJwcmVmZXJyZWRfdXNlcm5hbWUiOiAiZmVhbmlsIiwgInNjb3BlcyI6IFsicmVhZCIsICJ3cml0ZSIsICJlbWFpbCIsICJwcm9maWxlIl0sICJ2ZXJzaW9uIjogIjEuMi4wIiwgInN1YiI6ICI1YzEwZjY2ZjJkNjM5MGI3MDY2MmM5MTRhYWU3ZWY3OSIsICJmaWx0ZXJzIjogW10sICJpc19yZXN0cmljdGVkIjogZmFsc2UsICJlbWFpbF92ZXJpZmllZCI6IHRydWUsICJlbWFpbCI6ICJmZWFuaWxAYXhpbS5vcmciLCAibmFtZSI6ICJGZWFuaWwgUGF0ZWwiLCAiZmFtaWx5X25hbWUiOiAiIiwgImdpdmVuX25hbWUiOiAiIiwgImFkbWluaXN0cmF0b3IiOiB0cnVlLCAic3VwZXJ1c2VyIjogdHJ1ZX0.CX1S0QGrWKEPOHC8kUzGcvW8Ky04RCA8vU8WJrZURSw',
+    'expires_in': 3600,
+    'scope': 'read write email profile',
+    'token_type': 'JWT'}
+
+.. note:: When you get a JWT using ``client_credentials`` you don't get a
+   refresh token.  You're just expected to make a new call with your client
+   credentials.
+
+Check to see if a JWT is Expired
+********************************
+
+.. code-block::
+
+   import jwt
+
+   # See above examples for how to get a JWT token
+   jwt_token = token_request.json()['access_token']
+
+   try:
+       jwt.decode(jwt_token, "secret", audience="lms-key", algorithms=['HS256'])
+   except jwt.ExpiredSignatureError:
+       # Signature has expired
+
+Refresh a JWT Using a Refresh Token
+***********************************
+
+.. code-block::
+
+   import requests
+
+   # See above examples for how to get a JWT token with a refresh token
```

Comment:
1. Doesn't this comment describe this example, and not anything above?
2. How does the user get a refresh token? Is that a missing section?

Reply:
This comment describes getting a JWT token from a username and password.  Because you can't get a refresh token when using a confidential client as far as I can tell.

Reply:
The main point is that this example, aptly titled "Refresh a JWT Using a Refresh Token", relies on a `refresh_token`, but the document is missing a section (or note) detailing how I might retrieve such a token.

It seems like this code comment should point me to these instructions.

Reply:
Fair point, I added a longer comment that's hopefully more clear.

Reply:
Is the following what you are trying to convey?
```suggestion
   # See "Get a JWT with a Username and Password" for how to get a refresh token
```
If so, it is a little fuzzy because "with" (in your comment) and "using" (in the title of this section) are so similar that I think it was where I was getting confused. Consider my language, or some other update. Thanks.